include/inexor/vulkan-renderer/camera.hpp

```diff
@@ -1,127 +1,147 @@
 #pragma once
 
 #include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+#include <algorithm>
+#include <array>
+#include <vector>
 
 namespace inexor::vulkan_renderer {
 
-enum class CameraType { LOOKAT, FIRSTPERSON };
+enum class CameraMovement { FORWARD, BACKWARD, LEFT, RIGHT };
+
+// TODO: Implement more camera types.
+enum class CameraType { CAMERA_TYPE_LOOKAT };
 
-/// @brief A RAII wrapper class for camera position and movement.
-/// @todo Refactor method naming!
 class Camera {
-    /// @brief Update the view matrix.
-    /// @note The view matrix should only be updated if necessary.
-    void update_view_matrix();
+private:
+    CameraType m_type = CameraType::CAMERA_TYPE_LOOKAT;
+    glm::vec3 m_position = glm::vec3(0.0f, 0.0f, 0.0f);
+    glm::vec3 m_front = glm::vec3(0.0f, 0.0f, 1.0f);
+    glm::vec3 m_right = glm::vec3(1.0f, 0.0f, 0.0f);
+    glm::vec3 m_up = glm::vec3(0.0f, 1.0f, 0.0f);
+    glm::vec3 m_world_up = glm::vec3(0.0f, 1.0f, 0.0f);
+    glm::mat4 m_view_matrix = glm::mat4();
+    glm::mat4 m_perspective_matrix = glm::mat4();
+
+    float m_yaw = 0.0f;
+    float m_roll = 0.0f;
+    float m_pitch = 0.0f;
+    float m_pitch_min = -89.0f;
+    float m_pitch_max = +89.0f;
+    float m_fov = 90.0f;
+    float m_zoom_step = 10.0f;
+    float m_fov_max = 90.0f;
+    float m_fov_min = 20.0f;
+    float m_rotation_speed = 1.0f;
+    float m_movement_speed = 2.0f;
+    float m_aspect_ratio = 1920.0f / 1080.0f;
+    float m_mouse_sensitivity = 0.005f;
+    float m_near_plane = 0.001f;
+    float m_far_plane = 1000.0f;
+
+    std::array<bool, 4> m_keys = {false, false, false, false};
+    bool m_update_needed = true;
+
+    void update_vectors();
+
+    void update_matrices();
+
+    [[nodiscard]] bool is_moving() const;
```

Comment:
Not public?

Reply:
The update methods should not be public because we don't want the programmer to update the matrices unnecessarily.
I will make `is_moving()` public.

Reply:
Oh wuuups i just mean `is_moving()`ofc